### PR TITLE
[docs] Add a website test for broken links

### LINKED ISSUE #360: Docs: implement automatic testing for broken links

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**
fixes #360

### CODE REVIEW

**Comment by VasilyStrelyaev** [comment]
It's not experimental anymore. We're just waiting for @AlexanderMoskovkin  to tweak test-cafe-build-bot a bit

**Comment by VasilyStrelyaev** [comment]
@MargaritaLoseva - fix these broken links pls

**Comment by MargaritaLoseva** [comment]
Broken links fixed in #493 pull request

**Comment by VasilyStrelyaev** [comment]
@DevExpress/testcafe-docs 
Please **review** but **don't merge**.
We just need to ensure that our approach is okay before @AlexanderMoskovkin begins tweaking the test bot to run this test

**Comment by VasilyStrelyaev** [comment]
In addition, this PR downcases all filenames and links for cross-platform compliance and removes a couple of broken links

**Comment by VasilyStrelyaev** [comment]
does anyone interested?

**Comment by MargaritaLoseva** [comment]
As for broken links, lgtm

File: Gulpfile.js [hunk 1]
```diff
@@ -136,6 +137,7 @@ var CLIENT_TESTS_SAUCELABS_SETTINGS = {
     timeout:   720
 };
 
+var server;
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: I suppose it's better to call `websiteServer`

File: Gulpfile.js [hunk 2]
```diff
@@ -396,13 +398,43 @@ gulp.task('build-website', ['prepare-website'], function (cb) {
...
+            if (result.broken) {
+                brokenLinks.push(result.url.resolved ? result.url.resolved : result.url.original);
+                fail = true;
+            }
+        },
+        page: function (error, pageUrl) {
+            if (brokenLinks.length > 0) {
+                util.log('URL: ' + pageUrl);
+                util.log(util.colors.red('Broken links:'));
+                while (brokenLinks.length > 0)
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: separate the `while` block with empty lines (before the block and after)

File: Gulpfile.js [hunk 3]
```diff
@@ -396,13 +398,43 @@ gulp.task('build-website', ['prepare-website'], function (cb) {
...
+                    util.log(util.colors.red(brokenLinks.pop()));
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: Maybe it's better to use `brokenLinks.shift()` to log errors in the order they were found

File: Gulpfile.js [hunk 4]
```diff
@@ -396,13 +398,43 @@ gulp.task('build-website', ['prepare-website'], function (cb) {
...
+
+    var siteChecker = new blc.SiteChecker({ }, {
+        link: function (result) {
+            if (result.broken) {
+                brokenLinks.push(result.url.resolved ? result.url.resolved : result.url.original);
+                fail = true;
+            }
+        },
+        page: function (error, pageUrl) {
+            if (brokenLinks.length > 0) {
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: you can check just `if (brokenLinks.length)`

File: Gulpfile.js [hunk 5]
```diff
@@ -396,13 +398,43 @@ gulp.task('build-website', ['prepare-website'], function (cb) {
...
+                util.log('URL: ' + pageUrl);
+                util.log(util.colors.red('Broken links:'));
+                while (brokenLinks.length > 0)
+                    util.log(util.colors.red(brokenLinks.pop()));
+                util.log();
+            }
+        },
+        end: function () {
+            server.close(cb(fail ? 'Broken links found!' : void 0));
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: the `server.close` function gets `callback` as an argument. But in fact you call it with the result of the `cb(...)` function execution (it's `undefined`). You should wrap `cb` calling with a function:
```js
server.close(function () {
    cb(fail ? 'Broken links found!' : void 0)
});
```

**Comment by VasilyStrelyaev** [comment]
FPR

File: Gulpfile.js [hunk 6]
```diff
@@ -136,6 +137,7 @@ var CLIENT_TESTS_SAUCELABS_SETTINGS = {
     timeout:   720
 };
 
+var websiteServer;
```
Comment (AlexanderMoskovkin) [inline Gulpfile.js]: oh, i forgot. We always initialize a variable with a default value. It's `null` in this case

**Comment by AlexanderMoskovkin** [comment]
lgtm in others

/cc @inikulin @churkin @kirovboris

**Comment by churkin** [comment]
lgtm

**Comment by inikulin** [comment]
👌

**Comment by VasilyStrelyaev** [comment]
@testcafe-build-bot \retest

**Comment by kirovboris** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
Hey! @DevExpress/testcafe-docs 
We have done with this.

**Comment by VasilyStrelyaev** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
Bot seem to work as expected, thanks to @AlexanderMoskovkin

**Comment by AlexanderMoskovkin** [comment]
There was added the `.travis-docs.yml` file and was changed the `.travis.yml` file since last review (when it was `lgtm` from all reviewers) so we can just look at these two files

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
@MargaritaLoseva Just a reminder: from now on, all PRs for docs must start from the **[docs]** prefix.

**Pull request merged**